app/cdap/components/GridTable/index.tsx

```diff
@@ -57,6 +61,22 @@ export default function GridTable() {
     },
   ]);
 
+  const { dataprep } = DataPrepStore.getState();
```

Comment:
should you be getting this state every time this is rerendered?

Reply:
yes we need updated store data in every render

Reply:
You're repeating this on line 134, that seems redundant as you're getting the latest state here on every render.